package.json

```diff
@@ -27,8 +27,12 @@
     "test:ember-compatibility": "ember try:each"
   },
   "dependencies": {
+    "@codemirror/basic-setup": "^0.20.0",
+    "@codemirror/lang-html": "^0.20.0",
+    "@codemirror/lang-xml": "^0.20.0",
```

Comment:
I asked arne and he responded `ye we havent figured out yet how to make them actualy optional` so I guess help is needed there

Reply:
I _think_ it should be possible by using the `dependencySatisfies` and `importSync` [macros](https://www.npmjs.com/package/@embroider/macros). So the debug component could try to import them and throw an assertion if they are not installed.

Reply:
yeah it's an ember thing, there's no tree shaking on unused ember components and it will complain that the debug component can't import these even if you don't use the component. this should be fixed when we can move to embroider in ember-rdfa-editor I think, but we're pretty far from that.

Reply:
No, I think it should still be possible in classic builds with the Embroider macros mentioned above. The idea is that the debug component uses importSync to import the dependencies instead of regular imports so ember-auto-import won't complain at build time about the missing dependency. Consumers using the debug component will then have to ensure the dependencies are installed, or they will see the errors at runtime when using the component (although it's probably better to throw a custom assertion with proper guidance). It works in my head at least, but I haven't actually tried it so I might be missing important details 😄.

Another option for classic builds would be to use the `treeFor` hooks in the addon to strip out the debug component if a certain build flag is set (or not set depending on what makes the most sense as a default), but that's less ideal since it won't work if you want to convert the addon to the v2 addon format (, so it's guaranteed tech debt).

Yet another option could be to move the debug component to a dedicated addon so users who want it can simply install it?